### ISSUE TITLE
Docs shouldn't say the `required` attribute is still necessary when `Validators.required` is used

### ISSUE THREAD

**Issue by JWess** [angular/angular#47805]
### Describe the problem that you experienced

The documentation says `<input>` should have the `required` attribute for accessibility reasons, even when its associated `FormControl` has `Validators.required`.

![image](https://user-images.githubusercontent.com/8697891/196523460-c7a37f45-a3b8-4ee0-aad3-d6469d80f37e.png)

#### Example 
Here is an example from my current project (Angular 14 and Material 14) where I am not explicitly adding the `required` attribute to my `<input>`:
![image](https://user-images.githubusercontent.com/8697891/196524148-1c2cb308-0d20-43ae-9a60-6ba182cb0500.png)
![image](https://user-images.githubusercontent.com/8697891/196524257-673e39a0-20f3-4d52-b277-d1d67fa67c17.png)

#### Resultant DOM:
![image](https://user-images.githubusercontent.com/8697891/196523045-e6addce4-fdf7-47a4-8f65-8b8570781c15.png)

### Enter the URL of the topic with the problem

https://angular.io/guide/form-validation#built-in-validator-functions

**Comment by JoostK** [comment]
The [`required`](https://github.com/angular/components/blob/7ede2d0413090841f573b138477a6a364edf87e5/src/material/input/input.ts#L90) and [`arIa-required`](https://github.com/angular/components/blob/7ede2d0413090841f573b138477a6a364edf87e5/src/material/input/input.ts#L96) attributes are applied as host binding by the `matInput` directive here, so the documentation is accurate without Angular Material (which should usually be left out-of-scope of Angular FW documentation, IMO).
